### PR TITLE
feat(angular): add support for additional configurations

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

We do not currently support any way to allow multiple configurations to be passed when building Angular apps.

## Expected Behavior

We should support additional configurations.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/angular/src/builders/webpack-browser/webpack-browser.impl.ts [hunk 1]
```diff
@@ -24,13 +25,56 @@ export type BrowserBuilderSchema = Schema & {
...
+  const selectedTarget =
+    workspaceConfig.projects[context.target.project].targets[
+      context.target.target
+    ];
+
+  const mergedAdditionalConfigurations = additionalConfigurations.reduce(
+    (config, configurationName) => {
+      if (!(configurationName in selectedTarget.configurations)) {
+        console.warn(
+          `Could not find configuration "${configurationName}" in "${context.target.target}" for project "${context.target.project}".`
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/builders/webpack-browser/webpack-browser.impl.ts:51]: Yeah actually that's a good idea

File: packages/angular/src/builders/webpack-server/webpack-server.impl.ts [hunk 2]
```diff
@@ -77,6 +78,16 @@ export function webpackServer(schema: Schema, context: BuilderContext) {
...
 
+  context.target.configuration = `${context.target.configuration}${
+    additionalConfigurations && additionalConfigurations.length > 0
+      ? `,${additionalConfigurations.join(',')}`
+      : ``
+  }`;
+  options.browserTarget = `${options.browserTarget
+    .split(':')
+    .slice(0, 2)
+    .join(':')}:${context.target.configuration}`;
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/builders/webpack-server/webpack-server.impl.ts:89]: yep, solid catch

**Comment by AgentEnder** [comment]
Should we do this through the flag/ executor option as currently implemented or bake it into readExecutorOptions in the core so that its supported for all executors?

**Comment by Coly010** [comment]
>  Should we do this through the flag/ executor option as currently implemented or bake it into readExecutorOptions in the core so that its supported for all executors?

We've had discussions of this already and it would too large a change to Core (caching and other things are also touched by it). 

Plus, configuration composition like this isn't necessarily a good thing or what we would want to promote as a best practice. That said, Angular CLI supports it so it makes sense to have some kind of support for this for Angular users. 

It should be contained to Angular and one executor however so that we don't start impacting other verticals. The concept of needing multiple configurations doesn't exist for React, JS, Next etc. It could also lead to a wide variety of bugs that all relate to merge order of configurations. 

For this, I think it's best we limit the scope of the support to one executor in Angular so that we can help support more Angular CLI users making the transition to Nx.

**Comment by AgentEnder** [comment]
Sure, if the reasoning is that we don't want to encourage this type of practice then that makes sense. I have seen some valid use cases for the kind of setup but it does get confusing. As long as we are not shying away just due to it being a larger change set than that is fine by me 👍

**Pull request closed without merging**